### PR TITLE
Explicit-comms house cleaning

### PR DESCRIPTION
**Pull request by madsbk**
This PR makes `dataframe_merge.py` standalone. The plan is to remove it completely but I would like to keep it around for benchmarking a little while longer. 

Also: 
- `dataframe_merge.py` and `dataframe_shuffle.py` are moved into a new sub-package `explicit-comms/dataframe` so that dataframe specific utility functions can go in there as well. 
- `explicit-comms/__init__.py` now doesn't import any sub-modules implicitly. 
- Function docs
- Type hints

### CODE REVIEW

File: dask_cuda/explicit_comms/comms.py [hunk 1]
```diff
@@ -2,40 +2,62 @@
...
-        Otherwise, initialize a new comms instance.
+def default_comms(client: Optional[Client] = None) -> "CommsContext":
+    """Return the default comms object
+
+    Creates a new default comms object if no one exist.
+
+    Parameters
+    ----------
+    client: Client, optional
+        If no default comm object exist, create the new comm on `client`
```
Comment (pentschev) [inline dask_cuda/explicit_comms/comms.py:22]:
```suggestion
        If no default comm object exists, create the new comm on `client`
```

**Review by pentschev — COMMENTED** [review]
I have a few minor comments, but generally looks good to me to the extent I'm capable of reviewing. It's probably good to have @rjzamora taking a more thorough look as well.

File: dask_cuda/benchmarks/local_cudf_shuffle.py [hunk 2]
```diff
@@ -35,9 +35,9 @@ def shuffle_dask(args, df, write_profile):
     return took
 
 
-def merge_explicit_comms(args, df):
+def shuffle_explicit_comms(args, df):
     t1 = clock()
-    wait(explicit_comms.dataframe_shuffle(df, column_names="data").persist())
+    wait(explicit_comms_shuffle(df, column_names="data").persist())
```
[the remark this answers is not in the document]
Comment (madsbk) [reply dask_cuda/benchmarks/local_cudf_shuffle.py]: Good point, changed it to use the full import path

**Review by rjzamora — APPROVED** [review]
This seems fine to me @madsbk - Thanks!

As far as I can tell there is no new logic here (the explicit-comms merge is still concatenating all input partitions per worker).  Is this correct?

**Comment by madsbk** [comment]
> As far as I can tell there is no new logic here (the explicit-comms merge is still concatenating all input partitions per worker). Is this correct?

Yes, this is only cleanup and reorganization

**Comment by madsbk** [comment]
@gpucibot merge

**Comment by madsbk** [comment]
@pentschev @rjzamora, thanks for the reviews!

**Pull request merged**